scripts/sonar-scan.sh

```diff
@@ -22,6 +22,12 @@
 # docker-compose ones are:
 #  - MRAA_SRC_DIR - path to mraa's git clone in the Docker container.
 
+# Check required environment variables and exit if they are not set
+MRAA_SRC_DIR=${MRAA_SRC_DIR:?value not provided}
+SONAR_PROJ_KEY=${SONAR_PROJ_KEY:?value not provided}
+SONAR_ORG=${SONAR_ORG:?value not provided}
+SONAR_TOKEN=${SONAR_TOKEN:?value not provided}
+
```

Comment:
@Propanu @alext-mkrs changed the sonar-scan.sh to only run if the environment has the proper variables. It will exit with code 1 and print a message to stderr if those env vars are not provided

For reference: http://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion